flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1748,3 +1934,180 @@ def get_mounts():
+        ``DestroyBlockDeviceDataset``.
+        """
+        # action = assertHasAction(
+        #     self, logger, RESIZE_BLOCK_DEVICE_DATASET, succeeded=True)
+        # all_such_actions = LoggedAction.of_type(
+        #     logger.messages, RESIZE_BLOCK_DEVICE_DATASET)
+        # self.assertEqual([action], all_such_actions)
+        # # Child actions are logged
+        # [unmount] = LoggedAction.of_type(logger.messages, UNMOUNT_BLOCK_DEVICE)
+        # [detach] = LoggedAction.of_type(logger.messages, DETACH_VOLUME)
+        # [resize] = LoggedAction.of_type(logger.messages, RESIZE_VOLUME)
+        # [attach] = LoggedAction.of_type(logger.messages, ATTACH_VOLUME)
+        # [mount] = LoggedAction.of_type(logger.messages, MOUNT_VOLUME)
+        # self.assertEqual([unmount, detach, resize], action.children)
+        1/0
+
+    @validate_logging(verify_run_log)
+    def test_run_grow(self, logger):
+        """
+        After running ``ResizeBlockDeviceDataset``, its volume has been
+        resized.
+
+        XXX: We manually format and mount the created volume to avoid also
+        exercising the ``CreateBlockDeviceDataset`` state change in the same
+        unit test.
+        """
+        # self.patch(blockdevice, "_logger", logger)
+
+        # node = u"192.0.2.3"
+        # dataset_id = uuid4()
+        # api = loopbackblockdeviceapi_for_test(self)
+        # volume = api.create_volume(
+        #     dataset_id=dataset_id, size=REALISTIC_BLOCKDEVICE_SIZE
+        # )
+        # volume = api.attach_volume(volume.blockdevice_id, node)
+        # device = api.get_device_path(volume.blockdevice_id)
+        # mountroot = mountroot_for_test(self)
+        # mountpoint = mountroot.child(unicode(dataset_id).encode("ascii"))
+        # mountpoint.makedirs()
+        # check_output([b"mkfs", b"-t", b"ext4", device.path])
+        # check_output([b"mount", device.path, mountpoint.path])
+
+        # deployer = BlockDeviceDeployer(
+        #     hostname=node,
+        #     block_device_api=api,
+        #     mountroot=mountroot,
+        # )
+        # change = ResizeBlockDeviceDataset(volume=volume, size=REALISTIC_BLOCKDEVICE_SIZE * 2)
+        # self.successResultOf(change.run(deployer))
+
+        # expected_volume = volume.set(size=REALISTIC_BLOCKDEVICE_SIZE * 2)
+
+        # self.assertEqual([expected_volume], api.list_volumes())
+        1/0
+
+    def test_run_shrink(self, logger):
+        """
+        After running ``ResizeBlockDeviceDataset``, its volume has been
+        shrunk.
+
+        XXX: This can be marked TODO and implemented later in FLOC-1503.
+        """
+        1/0
+
+    def test_resize_unmounted_volume(self):
+        """
+        ``ResizeBlockDeviceDataset`` can resize an attached but unmounted volume.
+
+        The result is a larger block device with an expanded filesystem
+        containing the data that was their before the resize.
+        """
+        # XXX This situation shouldn't be possible, right?
```

Comment:
It may come up.  It should probably be handled, maybe not as part of this branch though.  The end state should probably be that the filesystem is **mounted** though.  We don't currently have any reason to have a volume attached to a node but the filesystem not mounted (so this is a state to repair, not restore).